src/app/cli/src/commands/repository_delete_command.rs

```diff
@@ -46,6 +46,12 @@ impl RepositoryDeleteCommand {
 #[async_trait::async_trait(?Send)]
 impl Command for RepositoryDeleteCommand {
     async fn run(&mut self) -> Result<(), CLIError> {
+        if self.names.is_empty() && !self.all {
+            return Err(CLIError::usage_error(
+                "Specify a repository or use --all flag",
+            ));
+        }
+
```

Comment:
Please use (override) `Command::validate_args()` for this check

Reply:
Also, I think we should work through all the variations
1) no names & no `--all`
2) no names & `--all`
3) some names & no `--all`
4) some names & `--all`

---

I think some of these options are redundant and an error message should be provided for them

Reply:
done

Reply:
Hmm, I see that you did it for `DeleteCommand`, but I don't see it for `RepositoryDeleteCommand` -- do we need `match { ... }` here too?

Reply:
My bad. `delete` command got new validation for free :) I think it makes sense to keep it for consistency. Now added to `repo delete` as well